src/serializer/ResidualFunctions.js

```diff
@@ -169,6 +169,30 @@ export class ResidualFunctions {
     return factoryFunctionInfos;
   }
 
+  // Preserve residual functions' ordering from original source code.
+  // This is necessary to prevent unexpected code locality issues.
+  // [Algorithm] sort function based on following criterias:
+  // 1. source file alphabetically.
+  // 2. start line number.
+  // 3. start column number.
+  _sortFunctionByOriginalOrdering(functionEntries: Array<[BabelNodeBlockStatement, Array<FunctionInstance>]>): void {
+    functionEntries.sort((funcA, funcB) => {
+      const funcALocation = funcA[0].loc;
+      const funcBLocation = funcB[0].loc;
+      if (!funcALocation || !funcBLocation || !funcALocation.source || !funcBLocation.source) {
+        // Preserve the current ordering if there is no source location information available.
```

Comment:
Is this expected to ever happen? If not, consider an invariant instead.

Reply:
yeah, it is an invariant originally. 
However, I found it failed all the react compiler tests. I do not know why but all the AST node of the react compiler tests do not have location so it has to be check here. @trueadm, do you know why functions in react compiler tests do not location information from babel?

Reply:
Do you mean the JSX Babel nodes created by the serializer? I'm not sure how we can best add source to them.